CMakeLists.txt

```diff
@@ -317,7 +334,9 @@ if (MSVC)
 #$(SolutionDir)<toolset>msvc,<variant>release:<linkflags>/OPT:ICF=5
 #$(SolutionDir)<toolset>msvc,<variant>release:<linkflags>/OPT:REF
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-	add_definitions(-Wno-c++11-extensions)
+	if (abi STREQUAL "c++03")
```

Comment:
my understanding is that the only difference is that more defect-reports have been resolved and TR1 was included in the std::tr1 namespace. I think it's still commonly referred to as c++98 though, since that's still the version of the ISO standards document, maybe both strings could be added as synonyms?

Perhaps I'm not quite understanding the motivation here though. The issue is that boost changes its ABI depending on whether it's built with a c++11 compiler or not, right? So, in order to stay ABI compatible, libtorrent must be built with the same version of the standard, so it must be configurable? (would you mind mentioning this in building.rst, or somewhere appropriate too?)